src/query/string.js

```diff
@@ -1,134 +1,161 @@
-import * as helpers from '../helpers';
-
-const SqlString = {};
-export { SqlString as default };
-
-SqlString.escape = function(val, timeZone) {
-  // Can't do require on top of file because Raw has not yet been initialized
-  // when this file is executed for the first time.
-  const Raw = require('../raw')
+/*eslint max-len: 0, no-var:0 */
+
+export const charsRegex = /[\0\b\t\n\r\x1a\"\'\\]/g; // eslint-disable-line no-control-regex
+export const charsMap = {
+  '\0': '\\0',
+  '\b': '\\b',
+  '\t': '\\t',
+  '\n': '\\n',
+  '\r': '\\r',
+  '\x1a': '\\Z',
+  '"': '\\"',
+  '\'': '\\\'',
+  '\\': '\\\\'
+}
 
-  if (val === null || val === undefined) {
-    return 'NULL';
+function wrapEscape(escapeFn) {
+  return function finalEscape(val, ctx = {}) {
+    return escapeFn(val, finalEscape, ctx)
   }
+}
 
-  switch (typeof val) {
-    case 'boolean': return (val) ? 'true' : 'false';
-    case 'number': return val+'';
+export function makeEscape(config = {}) {
+  const finalEscapeDate = config.escapeDate || dateToString
+  const finalEscapeArray = config.escapeArray || arrayToList
+  const finalEscapeBuffer = config.escapeBuffer || bufferToString
+  const finalEscapeString = config.escapeString || escapeString
+  const finalEscapeObject = config.escapeObject || escapeObject
+  const finalWrap = config.wrap || wrapEscape
+
+  function escapeFn(val, finalEscape, ctx) {
+    if (val === undefined || val === null) {
+      return 'NULL';
```

Comment:
They run after, basically if `.toString` is used, this is the escaping that happens